src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/DocumentSymbols/RemoteDocumentSymbolService.cs

```diff
@@ -37,7 +37,8 @@ protected override IRemoteDocumentSymbolService CreateService(in ServiceArgs arg
             .GetGeneratedDocumentAsync(cancellationToken)
             .ConfigureAwait(false);
 
-        var csharpSymbols = await ExternalHandlers.DocumentSymbols.GetDocumentSymbolsAsync(generatedDocument, useHierarchicalSymbols, cancellationToken).ConfigureAwait(false);
+        var csharpSymbols = await ExternalHandlers.DocumentSymbols.GetDocumentSymbolsAsync(
+            generatedDocument, useHierarchicalSymbols, supportsVSExtensions: true, cancellationToken).ConfigureAwait(false);
```

Comment:
While this addresses the `[Obsolete]` error, it's not really the correct and complete fix. Co-hosting will eventually run in VS Code, where this value should be set to `false`. A complete fix needs the following:

1. Update `IRemoteDocumentSymbolService.GetDocumentSymbolsAsync(...)` to take a `bool supportsVSExtensions` parameter.
2. Update the signature of `RemoteDocumentSymbolService.GetSymbolsAsync(...)` to take the new parameter and pass it along to `ExternalHandlers.DocumentSymbols.GetDocumentSymbolsAsync(...)`.
3. Update `CohostDocumentSymbolEndpoint.GetRegistrations(...)` to retrieve and store the value of `VSInternalClientCapabilities.SupportsVisualStudioExtensions`.
4. Update `CohostDocumetnSymbolEndpoint.HandleRequestAsync(...)` to pass the stored value to `IRemoteDocumentSymbolService.GetDocumentSymbolsAsync(...)`.

I think that's everything that needs to be updated in product code, though I didn't look through tests. It should be a super-mechanical change, and I'd prefer implementing it in this PR rather than leaving a bug to fix later when co-hosting is used in VS Code.

Reply:
I understand how to do that. The work I mentioned above "Hardcoding to true for now until we plumb VSInternalClientCapabilities through init of the remote client (see https://github.com/dotnet/razor/issues/11102)" was going to me an immediate follow-up to the current completion PR and would provide correct data here. My preference is still not to do that work in this PR.

Reply:
In that case, please add a TODO comment above the hard coding with a link to the issue. That way it's harder to miss this when #11102 is implemented.

In addition, what's stopping us from implementing #11102? That seems like a straightforward change. Would it be helpful if @davidwengier or I went ahead and did that?

Reply:
I can do it. I need these changes to go in and merge for #11096

Reply:
https://github.com/dotnet/razor/pull/11121

Reply:
I think you implemented my suggestion above instead of #11102.

Reply:
Yup! Didn't even read the issue but I'll see about doing #11102 as well

Reply:
> In that case, please add a TODO comment above the hard coding with a link to the issue. That way it's harder to miss this when #11102 is implemented.
> 
> In addition, what's stopping us from implementing #11102? That seems like a straightforward change. Would it be helpful if @davidwengier or I went ahead and did that?

There is nothing stopping it other than that I haven't gotten to it yet. Upgrading the packages to have all tests passing in the cohosting completion PR seemed like a more important task. That would be the next thing I would work on, but if one of you gets to it first that's fine as well.